COPYING

```diff
@@ -10,6 +11,8 @@ modify file(s), you may extend this exception to your version of the file(s),
 but you are not obligated to do so. If you do not wish to do so, delete this
 exception statement from your version.
 
+Some files have their own licenses. Take a look at the AUTHORS file.
```

Comment:
This statement (and all related stuff) contradicts the explanations from GPL FAQ in my eyes.
Entire Program (all its sources including borrowed from another Programs) should be distributed under the same license. If some Program license disallow you to distribute it (or its part) under GPL (i.e. Program has GPL incompatible license) you cannot include it (or it's part) into Program that is distributed under GPL.

Reply:
>If some Program license disallow you to distribute it (or its part) under GPL (i.e. Program has GPL incompatible license) you cannot include it (or it's part) into Program that is distributed under GPL.

Yes. But I wasn't refering to that. I was refering to taking eg a MIT licensed source and using it. We can use it in GPL. But also that file has remained in MIT license, since we haven't explicitly relicensed it (and I don't know if relicensing was possible).

>This statement (and all related stuff) contradicts the explanations from GPL FAQ in my eyes.

Is there a direct FAQ answer or is it from several ones?

Reply:
Also `AUTHORS` has some non-source files.

Reply:
https://www.gnu.org/licenses/gpl-faq.en.html#WhatDoesCompatMean

Reply:
When we distribute qBittorrent we do that as a combination of code and that combination is released under the same GNU GPL version. However the 3rd party files, when taken individually out of the project, can **also** be used under their original license. If they are modified by us we could make them GPLv2+ only, but it would need to be an explicit action. I don't think any contributor has done that, thus those modifications can be used under the original license too.
Well, that's my interpretation of all that.

Reply:
Maybe change the last sentence to:
```
Some 3rd party source files are under different licenses. Also non-source files are under their own licenses. See the AUTHORS file for details. It also contains some copyright info about the aforementioned files.
```

?

Reply:
>However the 3rd party files, when taken individually out of the project, can also be used under their original license.

Yes, if they are taken as part of their original program/project.
> If they are modified by us we could make them GPLv2+ only, but it would need to be an explicit action. 

Don't understand what explicit action you mean.

Let's look at an example of a lifecycle segment of arbitrary qBittorrent.
1. Someone got a qBittorrent v1 under GPL and "another app" under a compatible license, took the `file1` file from "another app", added it to qBittorrent, and then distributed a modified version of qBittorrent (v2) under the GPL. Now there are, as it were, two versions of the same file `file1`, distributed under different license (in different projects).
2. Someone got qBittorrent v2 under the GPL, modified it, including by modifying the `file1` file, and then distributed the modified version of qBittorrent (v3) under the GPL. Now a modified version of the `file1` file exists only as part of a program (qBittorrent v3) licensed under the GPL.

In any case, it seems to me incorrect to say about some qBittorrent files that they have their own license (unless they belong to those files that, in principle, do not fall under the GPL).

Reply:
It's like the Qt where it has some modules under GPLv3 only. The GPLv2+ Qt modules continue to be under the GPLv2+ license if they are used separately from the GPLv3 modules.

>Don't understand what explicit action you mean.

Let's say I modified a 3rd party file that I copied from a compatible project. If I  didn't explicitly state* that my modifications to this file are under GPLv2+, then they are under the license that is mentioned at the top of the file.

*I could make such statement by adding a modification to the license block of the 3rd party file.

But the details of this issue don't matter much to us. Any person wishing to copy said file should do their due diligence and decide under which license the modified file is and what to do with it.
So in order for this to go forward how about I change the offending sentene to simply:
```See also the AUTHORS file```?

Reply:
>So in order for this to go forward how about I change the offending sentene to simply:
```See also the AUTHORS file```?

Ok, let's go with it.

Reply:
done

Reply:
PR is approved to go further. But I will still add something else, as food for thought.

When someone makes public contribution in GPL project, it is obvious that they expect their work to be distributed under the terms of the GPL. For example, someone may not want their work to be used in closed source programs, etc. Putting a declaration that some files have a different license, we introduce confusion. Ideally, we should only mention that such and such files came from such and such projects that have such and such licenses, and not create the false impression that we distribute them under their own licenses.
>It's like the Qt where it has some modules under GPLv3 only. The GPLv2+ Qt modules continue to be under the GPLv2+ license if they are used separately from the GPLv3 modules.

Qt explicitly distributes some modules separately as an option. qBittorrent doesn't distribute any parts separately.
>Let's say I modified a 3rd party file that I copied from a compatible project. If I didn't explicitly state that my modifications to this file are under GPLv2+, then they are under the license that is mentioned at the top of the file.

Seems you still don't understand. When you contribute to qBittorrent, you modify qBittorrent's files under the terms of qBittorrent's license (even if some of these files were ever borrowed from a third-party project that has a different license).